### PR TITLE
fix dead cat relationships + patrol fixes

### PR DESCRIPTION
**Pull request by selkirks**
a fix in gen_train_deputydoubt1 in which a leader was regarded as a deputy. reported here: https://github.com/ClanGenOfficial/clangen/issues/2001

a fix in which you were able to view relationships of dead cats. reported here: https://github.com/ClanGenOfficial/clangen/issues/1971

a fix in which a loner medicine cat generated as an adolescent. reported here: https://github.com/ClanGenOfficial/clangen/issues/2015

### CODE REVIEW

**Review by Tiritirimatangi — COMMENTED** [review]
The patrol fixes look good! I can't comment on the coding fix, sorry

**Review by sedgestripe — APPROVED** [review]

File: resources/dicts/patrols/new_cat_welcoming.json [hunk 1]
```diff
@@ -992,7 +992,7 @@
                 "weight": 20,
                 "outsider_rep": 0,
                 "new_cat": [
-                    ["meeting", "loner"]
+                    ["status:medicine cat","meeting", "loner"]
```
Comment (Lixxis) [inline resources/dicts/patrols/new_cat_welcoming.json]: This line will not change the status of the cat. A cat outside the clan will get the status "loner", even with that added tag. I tested it out and looked into the save of the cat. To assure that this cat get the "medicine cat" after deciding to join the clan we would have to look at the background and assigning a status according to that.

Comment (selkirks) [reply resources/dicts/patrols/new_cat_welcoming.json]: do i assign one of the med cat backstories?

Comment (Lixxis) [reply resources/dicts/patrols/new_cat_welcoming.json]: In my test run it assigned the "wandering_healer2" backstory.

**Review by Ryos00 — APPROVED** [review]

**Pull request merged**